examples/audio-queue-squarewave.rs

```diff
@@ -36,7 +36,7 @@ fn main() {
         // default sample size 
         };
 
-    let device = audio_subsystem.open_queue::<i16>(None, &desired_spec).unwrap();
+    let device = audio_subsystem.open_queue::<i16, Option<&str>>(None, &desired_spec).unwrap();
```

Comment:
Yeah turns out type implication for generics is an all or nothing deal.  If you have to specify any type in order to get the call to work you have to specify all the types.  I tried doing <i16, ..> but that didn't work.  Most of the functions I modified didn't have generics before, so they were improved, but these functions did have generics before and in places where it was necessary to specify those generics it also became necessary to specify the Option types.  So for functions where they already had generics that needed to be specified it really is a tossup whether or not this is an improvement.  I'll defer to your judgement on whether or not we should keep this pattern.